src/theme/index.ts

```diff
@@ -52,16 +52,12 @@ type SCWUITheme = typeof theme & {
   linkComponent?: unknown
 }
 
-type RecursivePartial<T> = {
-  [P in keyof T]?: RecursivePartial<T[P]>
-}
-
 /**
  * Will extend theme with new theme properties
  * @param {SCWUITheme} baseTheme the theme you want to extend from, by default it is set to light theme
  * @param {RecursivePartial<SCWUITheme>} extendedTheme the properties of a new theme you want to apply from baseTheme
  */
-const extendTheme = (extendedTheme: RecursivePartial<SCWUITheme>) =>
+const extendTheme = (extendedTheme: Record<string, unknown>) =>
```

Comment:
Why ? 🤔

Reply:
@chambo-e We can override the theme with any type of variable. For instance if I want to store `setTheme` in my theme, I need to be able to do it without having type error saying I am not allowed to add it.

Reply:
That's so error prone, our theme should match our predefined scheme, if an user want to set a custom theme he can do it through emotion, not through us

emotion.d.ts
```ts
type MyCustomTheme = {
  setTheme: () => void
}

export interface Theme extends ScwUITheme & MyCustomTheme {}
```tsx

index.tsx
```
import { theme } from '@scaleway/ui'
import { ThemeProvider } from '@emotion/react'

const App = () => {
  const theme: Theme = { ...theme, setTheme: () => { switch() } }

  return (
    <ThemeProvider theme={theme}>{...}</Theme>
  )
}
```